csharp/ql/src/semmle/code/csharp/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -953,7 +953,7 @@ private module OutNodes {
         additionalCalls = false and
         call.(ImplicitDelegateDataFlowCall).isArgumentOf(csharpCall(_, cfn), _)
         or
-        additionalCalls = true and call = TTransitiveCapturedCall(cfn)
+        additionalCalls = true and call = TTransitiveCapturedCall(cfn, _)
```

Comment:
I believe this should be `TTransitiveCapturedCall(cfn, n.getEnclosingCallable())`

Reply:
And would be nice with a test for that case as well, i.e., two nested lambdas that both write to a captured variable, but only one of them writes a tainted value.

Reply:
In theory, I think you're right. In practice, I was unable to find a test case where this actually made a difference. I think this is because `OutNode.getCall()` is not used in practice in the dataflow libraries. Instead [DataFlowDispatch::Cached::getAnOutNode(DataFlowCall call, ReturnKind kind)](https://github.com/Semmle/ql/blob/master/csharp/ql/src/semmle/code/csharp/dataflow/internal/DataFlowDispatch.qll#L215-L227) associates `OutNode`s with `DataFlowCall`s in such a way that the runtime target is respected. I could be wrong, though! :slightly_smiling_face:

In any case, I've made the change as suggested, and added a test case for captured out variables (whose results did not change after this line was modified).

Reply:
You are right; in fact it is because the return kind, `ImplicitCapturedReturnKind`, explicitly carries the captured variable that the two updates do not get mixed up.